Hotfix/additional form options error case

carnage opened the pull request:
Fixes an additional case of the previous bug where composed objects overwrite options specified in other annotations

carnage committed:
Added date select filter and updated element to use it

carnage committed:
Added date time select filter and updated element to use it

carnage committed:
Added new filters to filter manager

carnage committed:
Added licence

carnage committed:
Abstracted date select filter logic to reuse for month select

carnage committed:
Added unit test for month select filter

carnage committed:
Added month select filter to manager and element

carnage committed:
Updated doc blocks

ojhaujjwal committed:
standard hydrator

ojhaujjwal committed:
factory for StandardHydrator

gabrielsch committed:
Fixing prepended `NotEmpty` validator

carnage committed:
Test case which covers additional use cases

carnage committed:
Fixed bug

jaapio commented:
please add a new testcase for this fix, rather then changing an existing one.

keradus commented on tests/ZendTest/Form/Annotation/AnnotationBuilderTest.php:
```diff
@@ -198,7 +199,7 @@ public function testOptionsAnnotationAndComposedObjectAnnotation($childName)
         $child = $form->get($childName);
 
-        $target = $child->getTargetElement();
+        $target = ($child instanceof Collection)? $child->getTargetElement() : $child;
```
I would add a space before that `?`

carnage commented:
@jappio This fix is for another occurance of the bug that the test is for, it extends the test case via a data provider, so in this instance I don't think a separate test case is warrented.

jaapio commented:
Changing a test is always a bad thing. The test might have the same intension as the issue your are testing. But somehow you found a bug in the behavior, so you need te be sure you don't break anything else. Leaving existing tests as they are will help you checking this.

Ocramius commented:
Agree with @jaapio here: can a new test be written instead?

carnage commented:
ok I'll c+p the test into a new method.

Linking #7108 for reference

Martin-P committed:
Failing testcases for issue 2589

weierophinney committed:
DBA inifile is not supported for caching

We recently noticed failures occuring for the DbaInifile cache storage tests.
The errors reported indicated an inability to perform `dba_replace()`
operations, with no indication of why this was.

On digging through the various DBA documentation on php.net, I discovered that
the `dba_replace` and `dba_insert()` functions are only reliable on GDBM and
QDBM handlers; any other handler may result in unexpected or unintended
behavior.

I have tested against flatfile, GDBM, DB4, and GDBM, and none of these show any
issues; however, inifile reliably has issues at this point. As such, we can no
longer support that version.

This patch raises an exception if the inifile handler is provided as a DBA
option, and the tests have been modified to ensure that the exception is raised.

carnage committed:
Updated to move new tests into separate test case

stefanotorresi committed:
add Zend\Mvc\HttpMethodListener

stefanotorresi committed:
add Zend\Mvc\Service\HttpMethodListenerFactory

stefanotorresi committed:
make Application::bootstrap() ready for HttpMethodListener integration

stefanotorresi committed:
refactor Application::bootstrap() tests

previous tests assertions were too specific and
prevented the introduction of additional default event listeners

ojhaujjwal committed:
renamed to MappingHydrator

weierophinney committed:
Merge branch 'feature/6709' into develop

Close #6709

weierophinney committed:
[#6709] Updated copyright

- to 2015

weierophinney committed:
Merge branch 'feature/6709' into develop

Update copyright in new files introduced in #6709

weierophinney committed:
Merge pull request #6712 from carnage/annotation-factory

Added form annotation builder factory

Conflicts:
	library/Zend/Mvc/Service/ServiceListenerFactory.php

weierophinney committed:
[#6712] Updated copyright year to 2015

weierophinney committed:
Merge branch 'feature/6712' into develop

Close #6712

weierophinney committed:
Merge pull request #6731 from bgotink/dev-filefilessize

fix for FilesSize validator usage with Input

weierophinney committed:
Merge branch 'feature/6731' into develop

Close #6731

weierophinney committed:
Merge pull request #6746 from nickpeirson/DI-RuntimeCompiler-Optimisation

Di runtime definition optimisation - Cache result of processing and inline the check

Conflicts:
	library/Zend/Di/Definition/RuntimeDefinition.php

weierophinney committed:
Merge branch 'feature/6746' into develop

Close #6746

weierophinney committed:
Merge pull request #6747 from nickpeirson/DI-DefinitionList-Optimisation

DI definition list optimisation - Cache classes upfront where possible to reduce iterations

weierophinney committed:
[#6747] Update to CS for consistency

- space around operators and between parens/opening braces
- use strict comparison when checking return value of assignment in a
  conditional

weierophinney committed:
Merge branch 'feature/6747' into develop

Close #6747

weierophinney committed:
Merge pull request #6752 from carnage/label-position-option

Added ability to specify label position as an element label option

Merge fixes test errors introduced in the original patch.

Conflicts:
	library/Zend/Form/View/Helper/FormRow.php

weierophinney committed:
Merge branch 'feature/6752' into develop

Close #6752

weierophinney committed:
Merge pull request #6754 from carnage/fix-annotation-builder-ordering

Added preserve_defined_order flag to annotation builder

weierophinney committed:
Merge branch 'feature/6754' into develop

Close #6754

Commit:
Add support for non 443 port

weierophinney committed:
Merge pull request #7235 from Lansoweb/master

Fixing Docblock return type for __invoke

weierophinney committed:
Merge branch 'feature/7235' into develop

Close #7235

weierophinney committed:
Merge pull request #7239 from samsonasik/hotfix/logger-exception

Fixes #7238 : Logger::log() exception message

Conflicts:
	tests/ZendTest/Log/LoggerTest.php

weierophinney committed:
Merge branch 'feature/7239' into develop

Close #7239

weierophinney committed:
Merge pull request #7241 from Martin-P/hotfix/curl-adapter

Removed non existing curl options

weierophinney committed:
Merge branch 'feature/7241' into develop

Close #7241

weierophinney committed:
Merge pull request #7245 from sandrokeil/navigation-abstract-service-factory

[Navigation] Introducing NavigationAbstractServiceFactory

weierophinney committed:
Merge branch 'feature/7245' into develop

Close #7245

weierophinney committed:
Merge pull request #7248 from researchgate/allow-curlopt-postfields

Remove CURLOPT_POSTFIELDS from set of invalid curl options

weierophinney committed:
Merge branch 'feature/7248' into develop

Close #7248

davidwindell committed:
Remove speech marks

weierophinney committed:
Merge pull request #6783 from carnage/continue-if-empty-annotation

Added continueIfEmpty annotation to Zend Form

weierophinney committed:
[#6783] Incorporate feedback

- Update copyright date
- Simplify original assigment of flag

weierophinney committed:
Merge branch 'feature/6783' into develop

Close #6783

weierophinney committed:
Added StandaloneExtensionManager as a drop-in extension plugin resolver

Essentially mimics the AbstractPluginManager, but does so as a
standalone implementation of ExtensionManagerInterface.

weierophinney committed:
Reader should use StandaloneExtensionManager by default

Modified getExtensionManager to lazyload a StandaloneExtensionManager
instead of ExtensionManager. This required an update to a test that was
testing integration of registerExtension with the EM instance itself.

weierophinney committed:
CS fix noticed when running php-cs-fixer

TRUE vs true

weierophinney committed:
Merge pull request #6267 from davidwindell/hotfix-address-commas

Support commas in names

weierophinney committed:
Merge branch 'feature/6267' into develop

Close #6267

weierophinney committed:
Merge pull request #6756 from samsonasik/feature/implement-todo-mime-part

implements @todo for setters/getters at Zend\Mime\Part

weierophinney committed:
Make syntax consistent with the framework

- We're migrating to `!` followed by a space
- If exceptions need interpolation, we use sprintf

weierophinney committed:
Merge branch 'feature/6756' into develop

Close #6756

weierophinney committed:
Merge pull request #6758 from samsonasik/imp.todo.adapter

add getAdapterInstance() that return adapter instance at Zend\Filter\Encrypt

weierophinney committed:
Adapt to be consistent with framework

- We're migrating to `!` followed by a space
- Use sprintf() when formatting exception messages that use
  interpolation
- Use magic constants where possible to reduce hard-coding

weierophinney committed:
Merge branch 'feature/6758' into develop

Close #6758

weierophinney committed:
Merge pull request #6790 from turrsis/hotfix/db-adapter-platform/sqlServerTest

Test/Db/Adapter/Platform/SqlServerTest restore_error_handler();

weierophinney committed:
Fix reported CS errors

- removed space before closing paren

weierophinney committed:
Merge branch 'feature/6790' into develop

Close #6790

improved-broccoli committed:
Disable input_filter_specs feature by default. Update tests.

weierophinney committed:
Merge pull request #6799 from Pittiplatsch/db_predicate_nextcombination

Zend\Db: Allow adding native predicates to sql queries, respecting nextPredicateCombineOrder

weierophinney committed:
Merge branch 'feature/6799' into develop

Close #6799

weierophinney committed:
Merge pull request #6827 from hissy/no-plural-form-lang

Fix inability to translate to languages which don't have plural forms

weierophinney committed:
Merge branch 'feature/6827' into develop

Close #6827

weierophinney committed:
Merge pull request #6856 from jameswithers/patch-1

Soap Connection Timeout

weierophinney committed:
Merge branch 'feature/6856' into develop

Close #6856

weierophinney committed:
Merge pull request #6931 from AGmakonts/patch-1

Added consts for left and right outer joins

weierophinney committed:
Merge branch 'feature/6931' into develop

Close #6931

weierophinney committed:
Merge pull request #6962 from gabrielsomoza/gabriel/zf-6960-whitelist-filter

New Whitelist / Blacklist Filter

weierophinney committed:
Updated copyright year

weierophinney committed:
Merge branch 'feature/6962' into develop

Close #6962

weierophinney committed:
Merge pull request #7016 from raum42/master

Adapt Json\Json for better prettyPrint support

weierophinney committed:
Update per php-cs-fixer

- Removed unneeded line following brace

weierophinney committed:
Merge branch 'feature/7016' into develop

Close #7016

weierophinney committed:
Merge pull request #7091 from Lansoweb/develop

Added Checkbox Console Prompt and its tests

weierophinney committed:
Updates recommended by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7091' into develop

Close #7091

weierophinney committed:
Merge pull request #7102 from neilime/patch-3

Add Kosovo phone number specifications

weierophinney committed:
Merge branch 'feature/7102' into develop

Close #7102

weierophinney committed:
Merge pull request #7104 from veewee/feature-bytes-filter

DataUnitFormatter filter

weierophinney committed:
Merge branch 'feature/7104' into develop

Close #7104

weierophinney committed:
Merge pull request #7120 from RalfEggert/feature/value-generator-array-comma

Added trailing comma for multiline arrays

weierophinney committed:
Merge branch 'feature/7120' into develop

Close #7120

weierophinney committed:
Merge pull request #7121 from ins0/fix/curl-ssl-verify

Feature: Added sslverifypeer to curl adapter

weierophinney committed:
Merge branch 'feature/7121' into develop

Close #7121

weierophinney committed:
Merge pull request #7122 from tklever/feature-paginator-iterator

Paginator Iterator

weierophinney committed:
Merge branch 'feature/7122' into develop

Close #7122

weierophinney committed:
Merge pull request #7125 from wesleywillians/zf2-7124

Adding ImportInterface for Zend\Feed\Reader\Reader

weierophinney committed:
Merge branch 'feature/7125' into develop

Close #7125
Fixes #7124

weierophinney committed:
Merge pull request #7128 from Martin-P/hotfix/session-validators

SessionManagerFactory + session validators via config makes validation always fail

weierophinney committed:
Fix CS issues reported by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7128' into develop

Close #7128

weierophinney committed:
Merge pull request #7152 from kusmierz/feature/4936-key-fail

Fixes #4936

weierophinney committed:
Merge branch 'feature/7152' into develop

Close #7152
Fixes #4936

weierophinney committed:
Merge pull request #7159 from Martin-P/hotfix/abstract-unicode-filters

Use getEncoding() instead of accessing options array directly

weierophinney committed:
Merge branch 'feature/7159' into develop

Close #7159
Fixes #7147

Logiqu2 committed:
fixed count error in config class

weierophinney committed:
Added file-level docblock.

weierophinney committed:
Merge pull request #7256 from Logiqu/config-count-fix

fixed count error in config class

weierophinney committed:
Merge branch 'feature/7256' into develop

Close #7256

weierophinney committed:
Merge pull request #7067 from marc-mabe/travis/cache

Travis: running more tests for cache component

Conflicts:
	.travis.yml

weierophinney committed:
Merge branch 'feature/7067' into develop

Close #7067

weierophinney committed:
Merge pull request #7163 from prolic/patch-10

Fix typos in inArray validators phpdoc

weierophinney committed:
Merge branch 'feature/7163' into develop

Close #7163

weierophinney committed:
Merge pull request #7177 from MarcelTO/master

Oracle quoting fix for values with single quotes

weierophinney committed:
Merge branch 'feature/7177' into develop

Close #7177

weierophinney committed:
Merge pull request #7178 from samsonasik/hotfix/ldap-composer

Fixes #7136 : add ext-ldap as required extension in Ldap\composer.json

weierophinney committed:
Merge branch 'feature/7178' into develop

Close #7178

weierophinney committed:
Merge pull request #7181 from hver/paginator-cache-interfaces

Remove hidden TaggableInterface dependencies from paginator

weierophinney committed:
Merge branch 'feature/7181' into develop

Close #7181
Fixes #7174

weierophinney committed:
Merge pull request #7182 from samsonasik/copyright-2015

update copyright year to 2015 that still uses 2013 in "develop"

weierophinney committed:
Merge branch 'feature/7182' into develop

Close #7182

weierophinney committed:
Merge pull request #7183 from samsonasik/hotfix/copyright-2015

update copyright year to 2015 that still uses 2013 in "master"

weierophinney committed:
Merge branch 'feature/7183' into develop

Close #7183

weierophinney committed:
Merge pull request #7184 from Victory/form-button-translate-content-param

FormButton $buttonContent parameter is now translated

weierophinney committed:
Merge branch 'feature/7184' into develop

Close #7184

weierophinney committed:
Merge pull request #7188 from Victory/fix-test-can-render-text-delimiters

fix test pattern in FormDateTimeSelectTest::testCanRenderTextDelimiters

weierophinney committed:
Merge branch 'feature/7188' into develop

Close #7188

weierophinney committed:
Merge pull request #7189 from stefanotorresi/test-nested-glob-patterns

add test for nested glob patterns

weierophinney committed:
Merge branch 'feature/7189' into develop

Close #7189

weierophinney committed:
Merge pull request #7190 from Martin-P/hotfix/pdo-connection-dsn-port-test

Test for array of PDO connection params

weierophinney committed:
Merge branch 'feature/7190' into develop

Close #7190

weierophinney committed:
Merge pull request #7194 from jaapio/docblock_fixes

fix some of the documentation errors

weierophinney committed:
Merge branch 'feature/7194' into develop

Close #7194

weierophinney committed:
Merge pull request #7201 from jaapio/cleanup/zfcode

Additional tests for zf-code

weierophinney committed:
Fix errors reported by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7201' into develop

Close #7201

weierophinney committed:
Merge pull request #7203 from marc-mabe/cache/fixMongoOptions

Cache: fix mongo options

weierophinney committed:
Perform argument validation

Tests indicated that an InvalidArgumentException should be thrown if the
`$resource` argument to `setResource` was not an array or
`MongoCollection`; however, no exception was thrown.

weierophinney committed:
Fix errors reported by php-cs-fixer

- Trailing whitespace

weierophinney committed:
Merge branch 'feature/7203' into develop

Close #7203

weierophinney committed:
Merge pull request #7210 from RalfEggert/bug/method-indention-from-class-reflection

Added fix to prevent method lines to be intended again and again

weierophinney committed:
Address php-cs-fixer errors

- Remove trailing line between closing braces.

weierophinney committed:
Merge branch 'feature/7210' into develop

Close #7210
Fixes #7209

weierophinney committed:
Merge pull request #7213 from chadicus/master

Error in FirePhp log writer

weierophinney committed:
Merge branch 'feature/7213' into develop

Close #7213
Fixes #7202

weierophinney committed:
Merge pull request #7215 from samsonasik/is_a

replace is_a() with instanceof

weierophinney committed:
Merge branch 'feature/7215' into develop

Close #7215

weierophinney committed:
Merge pull request #7225 from moderndeveloperllc/hotfix/FloatTest-testValidationFailures-fix

Fix for #7219 - Validation failure fr-CH in testValidationFailures

weierophinney committed:
Merge branch 'feature/7225' into develop

Close #7225

weierophinney committed:
Merge pull request #7231 from gianarb/feature/fix-zendtest-view-ext-intl

Skip tests in ZendTest\View if ext-intl is not installed

weierophinney committed:
Merge branch 'feature/7231' into develop

Close #7231

weierophinney committed:
Merge pull request #7232 from 27cm/iif

Remove unnecessary ternary operators

Conflicts:
	library/Zend/Db/Adapter/Driver/IbmDb2/Connection.php

weierophinney committed:
Merge branch 'feature/7232' into develop

Close #7232

weierophinney committed:
Merge pull request #7233 from rstgroup/master

Detect https on reversed proxy

weierophinney committed:
Merge branch 'feature/7233' into develop

Close #7233

weierophinney committed:
Merge pull request #7240 from stefanotorresi/feature/http-method-listener

Feature: Zend\Mvc\HttpMethodListener

weierophinney committed:
Address issues raised by php-cs-fixer

- single line after imports

weierophinney committed:
Merge branch 'feature/7240' into develop

Close #7240
Fixes #6385

weierophinney committed:
Merge pull request #7247 from jbenoit2011/import-input_filter_specs-from-zfcampus/zf-content-validation

Add input_filter_specs feature from zfcampus/zf-content-validation and associated tests

weierophinney committed:
Use specific return types when possible

weierophinney committed:
Spaces between @param and @return annotations are not necessary

weierophinney committed:
Merge branch 'feature/7247' into develop

Close #7247

weierophinney committed:
Merge pull request #6570 from ameir/smtp-envelope

Add ability to set SMTP envelope addresses

weierophinney committed:
Incorporate feedback

- Remove fluent interfaces
- Ensure docblocks are up-to-date
- Fix CS issues

weierophinney committed:
Update tests to remove fluent usage.

weierophinney committed:
Merge branch 'feature/6570' into develop

Close #6570

weierophinney committed:
Merge pull request #7056 from hschletz/inputfilter-refactor

Refactoring and bugfixes for InputFilter component

Conflicts:
	tests/ZendTest/InputFilter/BaseInputFilterTest.php

weierophinney committed:
Address issues raised by php-cs-fixer

- Anonymous function definitions

weierophinney committed:
Merge branch 'feature/7056' into develop

Close #7056

snapshotpl committed:
Clean reason phrase when status code is changed

snapshotpl committed:
Test comment

Victory committed:
Broken test for DateStepTime for Moscow time that spans 26-03-2011 - 27-03-2011 #7261

Victory committed:
DateStepValidator fix for Moscow Wintertime  - fixes #7261

Victory committed:
fix formating

Victory committed:
psr2 fixes

Victory committed:
psr2 fixes

ezimuel committed:
Merge branch 'feature/6419' of https://github.com/weierophinney/zf2 into weierophinney-feature/6419

ezimuel committed:
Merge branch 'weierophinney-feature/6419' into develop

weierophinney committed:
Use 5.3 array notation for compatibility

weierophinney committed:
Merge pull request #7259 from snapshotpl/feature/http-phrase

Override reason phrase by changing status code

weierophinney committed:
Merge branch 'feature/7259' into develop

Close #7259

weierophinney committed:
Merge branch 'hotfix/array-notation' into develop

Close #7265

ezimuel committed:
Merge branch 'weierophinney-hotfix/cache-dba-inifile' into develop

weierophinney committed:
Correct errors reported by php-cs-fixer

- in php-cs-fixer 1.5 on PHP 5.6

weierophinney committed:
Correct errors reported by php-cs-fixer

- in php-cs-fixer 1.5 on PHP 5.6

weierophinney committed:
Merge branch 'hotfix/phpcs-5.6' into develop

Close #7268

Victory committed:
fix typo and style

Victory committed:
remove use of special case and make more general for DateStep validator

Victory committed:
remove use of special case function

andrevus committed:
#7222 processJoins quote identifier error fix

carnage commented:
Build failures seem unrelated. Is this good to merge now @weierophinney

carnage committed:
Added preserve defined order as a factory option

carnage committed:
Added tests for invalid input values

carnage committed:
Added code to throw exception if filtering of an input is impossible

carnage committed:
CS changes

carnage committed:
CS fixes

Commit:
Email Validator vs IDN fixed

Martin-P committed:
Password value should never be rendered

Martin-P committed:
Fix for issue 7244 and applied same path for eventIntoColumn method

gammamatrix committed:
Fixed documentation on $path property

Fixed the docblock for `Zend\Mail\Transport::$path`

ezimuel committed:
Merge branch 'TomHAnderson-hotfix/deleteList' into develop

ojhaujjwal committed:
removed check for service existence

ojhaujjwal committed:
Moved factory to Stdlib namespace

Commit:
add Zend\Mail\MessageFactory

Commit:
mail writer use Zend\Mail\MessageFactory is $mail['mail'] is an array

Commit:
testsConstructWithMailAsArrayOptions for mail writer

Commit:
coding style, remove useless "use" statements

carnage committed:
Fixes indentation

carnage committed:
2014->2015

carnage committed:
2014->2015

carnage committed:
2014->2015

carnage committed:
2014->2015

fabiang committed:
Fix infinite loop when chaining peering service managers

lstrojny committed:
Fix reflection for one line docblocks

Commit:
update TLD list

weierophinney committed:
Merge pull request #7275 from Andrevus/zf7222

#7222 processJoins quote identifier error fix

weierophinney committed:
Merge branch 'feature/7222' into develop

Fixes #7222
Close #7275

weierophinney committed:
Merge pull request #7276 from carnage/add-options-to-annotation-builder-factory

Adds preserve defined order as a factory option

weierophinney committed:
Merge branch 'feature/7276' into develop

Close #7276

weierophinney committed:
Merge pull request #7277 from carnage/date-select-filter

Date select filter

weierophinney committed:
Incorporate feedback/perform code review

- Implemented early returns where possible
- Ensured constructors are present (to allow passing options via factory)
- Added `is` prefixes to various `null*()` methods
- Updated license year
- Removed class docblocks where they were not providing new information
- Used `@return self` for any fluent interfaces
- Ensure consistency of typehints
- Extracted `array_reduce` function to static class method (to allow opcode
  caching)

weierophinney committed:
Merge branch 'feature/7277' into develop

Close #7277

weierophinney committed:
Merge pull request #7281 from svyatoslav-kubakh/hotfix/email-validator-idn

Fix: Email Validator vs IDN

weierophinney committed:
CS fixes as indicated by php-cs-fixer

- Removes empty line between closing braces

weierophinney committed:
Merge branch 'feature/7281' into develop

Close #7281

weierophinney committed:
Merge pull request #7285 from gammamatrix/patch-3

Fixed documentation on $path property

weierophinney committed:
Merge branch 'feature/7285' into develop

Close #7285

weierophinney committed:
Merge pull request #7294 from Tauop/feature/log-writer-mail-factory

Feature/log writer mail factory

weierophinney committed:
Incorporated feedback for #7294

- Added data provider to test case
- Renamed test case to better describe behavior being tested
- Extracted setter name creation to a method for readability

weierophinney committed:
Merge branch 'feature/7294' into develop

Close #7294

weierophinney committed:
Merge pull request #7296 from fabiang/infinite-loop-sm-peering

Fix infinite loop when chaining peering service managers

weierophinney committed:
Incorporated feedback for #7296

- Continue early in loops.
- Ensure `$peeringServiceManager->serviceManagerCaller` is reset before
  returning from a loop.
- Remove `@covers` and `@uses` annotations which were raising errors.

weierophinney committed:
Merge branch 'feature/7296' into develop

Close #7296

weierophinney committed:
Merge pull request #7299 from lstrojny/bug/oneline-doc-blocks

Fix reflection for one line docblocks

weierophinney committed:
Merge branch 'feature/7299' into develop

Close #7299

weierophinney committed:
Merge pull request #7303 from Tauop/feature/validator-hostname-tld-update

update TLD list

weierophinney committed:
Merge branch 'feature/7303' into develop

Close #7303

weierophinney committed:
Merge pull request #6894 from ojhaujjwal/feature/standard-hydrator

standard hydrator idea

weierophinney committed:
Rename to DelegatingFactory

Per consensus on pull request, renamed from MappingHydrator to
DelegatingHydrator. Also, fixed CS issues, fixed license docblocks, and removed
entry from MVC service listener factory.

weierophinney committed:
Adapt the DelegatingHydratorFactory to work directly within the HydratorManager

Instead of pulling this from the application service manager, pull it
from the HydratorManager; this simplifies the factory, as it can just
pass its argument directly to the constructor.

weierophinney committed:
Fix CS issues reported by php-cs-fixer

- Remove unused use

weierophinney committed:
Merge branch 'feature/6894' into develop

Close #6894

weierophinney committed:
Merge pull request #7003 from gabrielsch/bugfix/input-not-empty

Fixing prepended `NotEmpty` validator

weierophinney committed:
Merge branch 'feature/7003' into develop

Close #7003

weierophinney committed:
Merge pull request #7171 from Martin-P/hotfix/hide-form-password-value

Reset password value on $form->prepare()

weierophinney committed:
Merge branch 'feature/7171' into develop

Close #7171
Fixes #7166

weierophinney committed:
Merge pull request #7262 from Victory/DateStepTime-for-moscow

DateStep time for moscow fixes #7261

weierophinney committed:
CS fixes during review

weierophinney committed:
Merge branch 'feature/7262' into develop

Close #7262
Fixes #7261

weierophinney committed:
Merge pull request #7282 from Martin-P/hotfix/log-writer-db-array-to-string-conversion

Fix: Array to string conversion in Zend\Log\Writer\Db

weierophinney committed:
CS fixes as reported by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7282' into develop

Close #7282
Fixes #7244

carnage commented:
reopened against develop #7309
